subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/ModifierAnnotationCategory.java

```diff
@@ -42,6 +43,9 @@
     )),
     OPTIONAL(ImmutableSet.of(
         Optional.class
+    )),
+    DIRECTORY_ONLY(ImmutableSet.of(
+        IgnoreDirectories.class
     ));
```

Comment:
I think this should either be called `IGNORE_EMPTY_DIRECTORIES` or if we want to be more general like the other categories, then `FILTERING`.

Reply:
Done.